graph/src/blockchain/substreams_block_stream.rs

```diff
@@ -186,13 +180,18 @@ fn stream_blocks<C: Blockchain, F: SubstreamsMapper<C>>(
     let mut skip_backoff = false;
 
     try_stream! {
+            let endpoint = client.firehose_endpoint()?;
+            let logger = logger.new(o!("deployment" => deployment.clone(), "provider" => endpoint.provider.to_string()));
+
         loop {
             info!(
                 &logger,
                 "Blockstreams disconnected, connecting";
                 "endpoint_uri" => format_args!("{}", endpoint),
```

Comment:
Not related to this PR, but for RPC providers we stopped logging URL's because they can contain sensitive information (passwords) and mostly just log provider names. Might be a good idea for firehose endpoints, too.

Reply:
Display has been explicitly implemented for the endpoint, only prints the provider